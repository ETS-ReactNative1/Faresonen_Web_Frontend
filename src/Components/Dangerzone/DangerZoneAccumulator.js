import React, { Component } from "react";
import { Feed, Icon } from "semantic-ui-react";

class DangerZoneAccumulator extends Component {
  constructor(props) {
    super(props);
    this.state = {
      data: {},
      loading: true
    };
  }
  render() {
    return (
      <Feed>
        <Feed.Event>
          <Feed.Label>
            <img src="http://fillmurray.com/200/200" />
          </Feed.Label>
          <Feed.Content>
            <Feed.Summary>
              <Feed.User>Elliot Fu</Feed.User> added you as a friend
              <Feed.Date>1 Hour Ago</Feed.Date>
            </Feed.Summary>
            <Feed.Meta>
              <Feed.Like>
                <Icon name="like" />
                4 Likes
              </Feed.Like>
            </Feed.Meta>
          </Feed.Content>
        </Feed.Event>
      </Feed>
    );
  }
}
export default DangerZoneAccumulator;
