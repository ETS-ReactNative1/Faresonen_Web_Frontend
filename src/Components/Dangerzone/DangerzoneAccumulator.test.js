import React from "react";
import { shallow } from "enzyme";
import DangerZoneAccumulator from "./DangerZoneAccumulator";
import testData from "../../Test/testData";

describe("<DangerzoneAccumulator />", () => {
  it("Should return a message element for the getInfoMessage()", () => {
    const elem = shallow(
      <DangerZoneAccumulator
        events={testData.yellowCardEvents}
        removeEvent={() => {}}
        removePlayer={() => {}}
      />
    );

    const message = elem.instance().getInfoMessage();
    expect(message.props.info).toEqual(true);
    expect(message.props.icon).toEqual("info");
    expect(message.props.header).toEqual("Informasjon");
  });

  it("should remove an event from state correctly", () => {
    const elem = shallow(
      <DangerZoneAccumulator
        events={[]}
        removeEvent={() => {}}
        removePlayer={() => {}}
      />
    );
    // Set state manually
    elem.state().data.events = testData.yellowCardAndPlayerEvents;
    expect(elem.state().data.events.length).toEqual(6);
    // Remove player from event
    const removeEvent = {
      player: {
        "@uri": "http://api.tv2.no/sport/resources/people/345617/",
        birthday: "1988-05-28T00:00:00+02:00",
        birthplace: "Pakrac",
        country: {
          "@uri": "http://api.tv2.no/sport/resources/countries/205/"
        },
        created: "2018-03-05T17:45:33+01:00",
        enetpulseId: "463298",
        firstname: "Zoran",
        height: "188",
        id: "345617",
        images: {
          "@uri": "http://api.tv2.no/sport/resources/people/345617/images/"
        },
        lastname: "Popovic",
        memberships: {
          "@uri": "http://api.tv2.no/sport/resources/people/345617/memberships/"
        },
        nifsId: "155223",
        role: {
          "@uri": "http://api.tv2.no/sport/resources/roles/1/"
        },
        weight: "81"
      },
      event: {
        "@uri": "http://api.tv2.no/sport/resources/events/15221417/",
        eventtype: {
          "@uri": "http://api.tv2.no/sport/resources/eventtypes/2/"
        },
        eventtime: "41",
        extendedeventtype: {
          "@uri": "http://api.tv2.no/sport/resources/extendedeventtypes/1200/"
        },
        goalsTeamA: "1",
        goalsTeamB: "0",
        id: "15221417",
        match: {
          "@uri": "http://api.tv2.no/sport/resources/matches/912485/"
        },
        person1: {
          "@uri": "http://api.tv2.no/sport/resources/people/345617/"
        },
        realTime: "2018-03-18T18:46:57+01:00",
        team: {
          "@uri": "http://api.tv2.no/sport/resources/teams/307/"
        }
      }
    };
    elem.instance().removeEventFromState(removeEvent);
    expect(elem.state().data.events.length).toEqual(5);

    // Try to remove an event with wrong player id
    expect(
      elem.instance().removeEventFromState({
        player: {
          id: 123
        }
      })
    ).toEqual(false);

    expect(elem.state().data.events.length).toEqual(5);

    const removeEmptyObject = () => {
      elem.instance().removeEventFromState({});
    };

    // Test to remove an event that does not exists
    expect(removeEmptyObject).toThrow();
  });
});
