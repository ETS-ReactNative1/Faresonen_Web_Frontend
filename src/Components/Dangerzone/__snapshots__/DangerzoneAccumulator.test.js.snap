// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DangerzoneAccumulator /> should render correctly 1`] = `
<div>
  <div
    className="ui icon info message"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <i
      aria-hidden="true"
      className="info icon"
      onClick={[Function]}
    />
    <div
      className="content"
    >
      <div
        className="header"
      >
        Informasjon
      </div>
      <p>
        Trykk her for mer informasjon om hvordan denne tjenesten virker.
      </p>
    </div>
  </div>
  <div
    className="ui feed"
  >
    <div
      className="ui info message"
    >
      <div
        className="content"
      >
        <div
          className="header"
        >
          Ingen gule kort registrert
        </div>
        <p>
          Det er ikke registrert noen gule kort. Når de blir registrert dukker de opp her, dersom den registrerte finnes i faresonen fra før.
        </p>
      </div>
    </div>
  </div>
</div>
`;
